src/handler/sessions/session.rs

```diff
@@ -140,32 +200,39 @@ impl Session {
         id_nonce_sig: &[u8],
         ephem_pubkey: &[u8],
         enr_record: Option<Enr>,
+        node_address: &NodeAddress,
+        sessions: &mut Sessions,
     ) -> Result<(Session, Enr), Discv5Error> {
```

Comment:
Is ``node_address: &NodeAddress,`` this the local nodes address or the remotes nodes address. We deal with both nodes id's. So it is a bit ambiguous which one it is. We could assume it is remote, but needing to assume things causes bugs when other people assume different things. Maybe use ``remote_node_address`` to stay consistent with the variables above?